packages/core/src/renderTexture/RenderTexturePool.js

```diff
@@ -0,0 +1,177 @@
+     * @returns {RenderTexture}
+     */
+    createTexture(realWidth, realHeight)
+    {
+        const baseRenderTexture = new BaseRenderTexture(Object.assign({
+            width: realWidth,
+            height: realHeight,
+        }, this.textureOptions));
+
+        return new RenderTexture(baseRenderTexture);
+    }
+
+    /**
+     * Gets a Power-of-Two render texture or fullScreen texture
+     *
+     * @protected
+     * @param {number} minWidth - The minimum width of the render texture in real pixels.
+     * @param {number} minHeight - The minimum height of the render texture in real pixels.
+     * @param {number} [resolution=1] - The resolution of the render texture.
+     * @return {PIXI.RenderTexture} The new render texture.
+     */
+    getOptimalTexture(minWidth, minHeight, resolution = 1)
+    {
+        let key = RenderTexturePool.SCREEN_KEY;
+
+        minWidth *= resolution;
+        minHeight *= resolution;
+
+        if (!this.enableFullScreen || minWidth !== this._pixelsWidth || minHeight !== this._pixelsHeight)
+        {
+            minWidth = nextPow2(minWidth);
+            minHeight = nextPow2(minHeight);
+            key = ((minWidth & 0xFFFF) << 16) | (minHeight & 0xFFFF);
+        }
+
+        if (!this.texturePool[key])
+        {
+            this.texturePool[key] = [];
+        }
+
+        let renderTexture = this.texturePool[key].pop();
+
+        if (!renderTexture)
+        {
+            renderTexture = this.createTexture(minWidth, minHeight);
+        }
+
+        renderTexture.filterPoolKey = key;
+        renderTexture.setResolution(resolution);
+
+        return renderTexture;
+    }
+
+    /**
+     * Gets extra texture of the same size as current renderTexture
+     *
+     * @param {number} [resolution] resolution, by default its the same as in current renderTexture
+     * @returns {PIXI.RenderTexture}
+     */
+    getFilterTexture(resolution)
+    {
+        const rt = this.renderer.renderTexture.current;
+
+        const filterTexture = this.getOptimalTexture(rt.width, rt.height, resolution || rt.baseTexture.resolution);
+
+        filterTexture.filterFrame = rt.filterFrame;
+
+        return filterTexture;
+    }
+
+    /**
+     * Place a render texture back into the pool.
+     * @param {PIXI.RenderTexture} renderTexture - The renderTexture to free
+     */
+    returnTexture(renderTexture)
+    {
+        const key = renderTexture.filterPoolKey;
+
+        renderTexture.filterFrame = null;
+        this.texturePool[key].push(renderTexture);
+    }
+
+    /**
+     * Alias for returnTexture, to be compliant with FilterSystem interface
+     * @param {PIXI.RenderTexture} renderTexture - The renderTexture to free
+     */
+    returnFilterTexture(renderTexture)
+    {
+        this.returnTexture(renderTexture);
+    }
+
+    /**
+     * Clears the pool
+     *
+     * @member {boolean} [destroyTextures=true] destroy all stored textures
+     */
```

Comment:
could this be:
`clear(destroyTextures = false)`

Reply:
no, its true by default, like in previous version.

Reply:
i dont know how to do that part, and its internal thing for a custom class, we can change it later.